### PR TITLE
Add KillContainerInPod in DockerManager

### PR DESCRIPTION
**Pull request by yujuhong**
This changes adds one of the functions that DockerManager needs to implement
the Runtime interface.

### CODE REVIEW

File: pkg/kubelet/dockertools/manager_test.go [hunk 1]
```diff
@@ -214,3 +215,127 @@ func TestGetPods(t *testing.T) {
 		t.Errorf("expected %#v, got %#v", expected, actual)
 	}
 }
+
+func apiContainerToContainer(c docker.APIContainers) kubecontainer.Container {
```
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager_test.go]: This function is no longer used, but seems handy so I keep it around.

File: pkg/kubelet/dockertools/manager_test.go [hunk 2]
```diff
@@ -214,3 +215,127 @@ func TestGetPods(t *testing.T) {
...
+		return kubecontainer.Container{}
+	}
+	return kubecontainer.Container{
+		ID:   types.UID(c.ID),
+		Name: dockerName.ContainerName,
+		Hash: hash,
+	}
+}
+
+func dockerContainersToPod(containers DockerContainers) kubecontainer.Pod {
```
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager_test.go]: This function is no longer used, but seems handy so I keep it around.

**Comment by vmarmol** [comment]
LGTM

File: pkg/kubelet/dockertools/manager_test.go [hunk 3]
```diff
@@ -214,3 +215,127 @@ func TestGetPods(t *testing.T) {
...
+	// Set all containers to ready.
+	for _, c := range fakeDocker.ContainerList {
+		manager.readinessManager.SetReadiness(c.ID, true)
+	}
+
+	if err := manager.KillContainerInPod(pod.Spec.Containers[0], pod); err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	// Assert the container has been stopped.
+	if err := fakeDocker.AssertStopped([]string{containerToKill.ID}); err != nil {
```
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager_test.go]: Changed to verify the *correct* container were stopped (before it only checked if the "stop" call existed).

**Comment by vmarmol** [comment]
Will need a rebase @yujuhong btw

**Comment by yujuhong** [comment]
@vmarmol, thanks for the reminder! Rebased.

**Comment by vmarmol** [comment]
LGTM

**Comment by yifan-gu** [comment]
LGTM!
We will need to figure out the paramater for `RunContainerInPod` and `KillContainerInPod`, maybe a runtime pod makes more sense for `KillContainerInPod`

**Comment by yujuhong** [comment]
@yifan-gu, this may not be needed anymore. See #7603

**Pull request merged**